packages/eas-cli/src/commandUtils/builds.ts

```diff
@@ -0,0 +1,68 @@
+import chalk from 'chalk';
+
+import { Build, BuildFragment, BuildStatus } from '../graphql/generated';
+import { BuildQuery } from '../graphql/queries/BuildQuery';
+import { appPlatformEmojis } from '../platform';
+import { ExpoGraphqlClient } from './context/contextUtils/createGraphqlClient';
+
+export async function ensureBuildExistsAsync(
+  graphqlClient: ExpoGraphqlClient,
+  buildId: string
+): Promise<void> {
+  try {
+    await BuildQuery.byIdAsync(graphqlClient, buildId);
+  } catch {
+    throw new Error(`Couldn't find a build matching the id ${buildId}`);
+  }
+}
+
+export async function fetchBuildsAsync(
+  graphqlClient: ExpoGraphqlClient,
+  projectId: string,
+  statuses?: BuildStatus[]
```

Comment:
I would use a `{ graphqlClient, projectId, statuses }` argument. This way, if we add more filters (like `platform: Platform[]`) it's much easier and readable.

Reply:
Updated